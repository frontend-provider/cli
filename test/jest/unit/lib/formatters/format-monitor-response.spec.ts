import stripAnsi from 'strip-ansi';

import {
  formatErrorMonitorOutput,
  formatMonitorOutput,
} from '../../../../../src/lib/formatters';

describe('formatErrorMonitorOutput', () => {
  it('maven monitor error', () => {
    const res = formatErrorMonitorOutput(
      'maven',
      {
        org: 'test-org',
        id: '123',
        path: 'src/lib/jens',
        licensesPolicy: {},
        uri: 'https://example.com/project',
        isMonitored: true,
        trialStarted: false,
      },
      {},
    );
    expect(stripAnsi(res)).toMatchSnapshot();
  });

  it('maven monitor error with --json', () => {
    const res = formatErrorMonitorOutput(
      'maven',
      {
        org: 'test-org',
        id: '123',
        path: 'src/lib/jens',
        licensesPolicy: {},
        uri: 'https://example.com/project',
        isMonitored: true,
        trialStarted: false,
      },
      { json: true },
    );
    expect(stripAnsi(res)).toMatchSnapshot();
  });

  it('npm with project name', () => {
    const res = formatErrorMonitorOutput(
      'npm',
      {
        org: 'test-org',
        id: '123',
        path: 'src/lib/jens',
        licensesPolicy: {},
        uri: 'https://example.com/project',
        isMonitored: true,
        trialStarted: false,
      },
      {},
      'custom-name',
    );
    expect(stripAnsi(res)).toMatchSnapshot();
  });
});

describe('formatMonitorOutput', () => {
  it('gradle scan with --all-sub-projects', () => {
    const monitorResult = {
      org: 'test-org',
      id: '123',
      path: 'src/lib/jens',
      licensesPolicy: {},
      uri: 'https://example.com/project',
      isMonitored: true,
      trialStarted: false,
    };
    const res = formatMonitorOutput(
      'gradle',
      monitorResult,
      { allSubProjects: true },
      undefined,
      7,
    );
    expect(stripAnsi(res)).toMatchSnapshot();
  });
  it('gradle scan without --all-sub-projects', () => {
    const monitorResult = {
      org: 'test-org',
      id: '123',
      path: 'src/lib/jens',
      licensesPolicy: {},
      uri: 'https://example.com/project',
      isMonitored: true,
      trialStarted: false,
    };
    const res = formatMonitorOutput('gradle', monitorResult, {}, undefined, 7);
    expect(stripAnsi(res)).toMatchSnapshot();
  });

  it('docker', () => {
    const monitorResult = {
      org: 'test-org',
      id: '123',
      path: 'src/lib/jens',
      licensesPolicy: {},
      uri: 'https://example.com/project',
      isMonitored: true,
      trialStarted: false,
    };
    const res = formatMonitorOutput(
      'deb',
      monitorResult,
      { docker: true },
      undefined,
      7,
    );
    expect(stripAnsi(res)).toMatchSnapshot();
  });

  it('npm without --all-projects (more projects were detected)', () => {
    const monitorResult = {
      org: 'test-org',
      id: '123',
      path: 'src/lib/jens',
      licensesPolicy: {},
      uri: 'https://example.com/project',
      isMonitored: true,
      trialStarted: false,
    };
    const res = formatMonitorOutput('npm', monitorResult, {}, undefined, 7);
    expect(stripAnsi(res)).toMatchSnapshot();
  });

  it('npm with --all-projects (more projects were detected)', () => {
    const monitorResult = {
      org: 'test-org',
      id: '123',
      path: 'src/lib/jens',
      licensesPolicy: {},
      uri: 'https://example.com/project',
      isMonitored: true,
      trialStarted: false,
    };
    const res = formatMonitorOutput(
      'npm',
      monitorResult,
      { allProjects: true },
      undefined,
      7,
    );
    expect(stripAnsi(res)).toMatchSnapshot();
  });

  it.todo('--json');
});
