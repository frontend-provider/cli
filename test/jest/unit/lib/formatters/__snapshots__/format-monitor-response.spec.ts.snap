// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formatErrorMonitorOutput maven monitor error 1`] = `
"
Monitoring src/lib/jens...



Detected 0 dependencies (no project created)"
`;

exports[`formatErrorMonitorOutput maven monitor error with --json 1`] = `"{\\"org\\":\\"test-org\\",\\"id\\":\\"123\\",\\"path\\":\\"src/lib/jens\\",\\"licensesPolicy\\":{},\\"uri\\":\\"https://example.com/project\\",\\"isMonitored\\":true,\\"trialStarted\\":false,\\"packageManager\\":\\"maven\\"}"`;

exports[`formatErrorMonitorOutput npm with project name 1`] = `
"
Monitoring src/lib/jens (custom-name)...



"
`;

exports[`formatMonitorOutput docker 1`] = `
"
Monitoring src/lib/jens...

Explore this snapshot at https://example.com/project

Tip: Detected multiple supported manifests (7), use --all-projects to scan all of them at once.

Notifications about newly disclosed issues related to these dependencies will be emailed to you.
"
`;

exports[`formatMonitorOutput gradle scan with --all-sub-projects 1`] = `
"
Monitoring src/lib/jens...

Explore this snapshot at https://example.com/project

Notifications about newly disclosed issues related to these dependencies will be emailed to you.
"
`;

exports[`formatMonitorOutput gradle scan without --all-sub-projects 1`] = `
"
Monitoring src/lib/jens...

Explore this snapshot at https://example.com/project

Tip: This project has multiple sub-projects (7), use --all-sub-projects flag to scan all sub-projects.

Notifications about newly disclosed issues related to these dependencies will be emailed to you.
"
`;

exports[`formatMonitorOutput npm with --all-projects (more projects were detected) 1`] = `
"
Monitoring src/lib/jens...

Explore this snapshot at https://example.com/project

Notifications about newly disclosed issues related to these dependencies will be emailed to you.
"
`;

exports[`formatMonitorOutput npm without --all-projects (more projects were detected) 1`] = `
"
Monitoring src/lib/jens...

Explore this snapshot at https://example.com/project

Tip: Detected multiple supported manifests (7), use --all-projects to scan all of them at once.

Notifications about newly disclosed issues related to these dependencies will be emailed to you.
"
`;
